src/NUnitFramework/framework/Internal/Execution/WorkItemQueue.cs

```diff
@@ -339,14 +339,60 @@ internal void Save()
         /// </summary>
         internal void Restore()
         {
-            Pause();
+            //Guard.OperationValid(State != WorkItemQueueState.Running, $"Attempted to restore state of {Name} while queue was running.");
```

Comment:
Is this a valid guard? Should it be uncommented or deleted?

Reply:
It's logically valid (in my mind) but breaks things if I use it. I think I'll keep it commented but add an explanation in front of it. At some point, someone will discover __why__ some queues end up running at a point where we are supposedly between shifts. My guess is that some other thread is getting in their and restarting things.

This is where I'd continue if I had time to do it by the way. We need one thread to detect end of shift and take care of starting next shift, restoring, etc. before any other shift tries to do the same. 

However, I think that overall this fixes a lot of things and is not any worse in the areas it doesn't fix.